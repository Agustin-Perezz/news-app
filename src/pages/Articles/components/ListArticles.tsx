import React from 'react';
import { Box, Grid, Typography } from '@mui/material';
import { NewsProps } from '../../../types/news-props';
import { Article } from './Article';
import { RenderController } from '../../../components/ui/RenderController';

interface Props {
  dataArticles?: NewsProps[];
  isLoading: boolean;
  isError?: string;
}

export const ListArticles: React.FC<Props> = ({
  dataArticles,
  isLoading,
  isError,
}) => {
  return (
    <RenderController
      isLoading={isLoading}
      isError={isError}
      data={dataArticles}
    >
      <Box sx={{ flexGrow: 1, my: 3 }}>
        <Typography variant="h5" mb={3} textAlign="center" color="snow">
          Results for search:
        </Typography>
        <Grid
          container
          spacing={{ xs: 2, md: 3 }}
          columns={{ xs: 0, sm: 8, md: 13 }}
          justifyContent="center"
          sx={{
            maxWidth: '1200px',
            margin: '0px auto 20px auto !important',
          }}
        >
          {dataArticles?.map((article) => (
            <Grid item xs={2} sm={4} md={3.5} key={article.uuid}>
              <Article dataArticle={article} />
            </Grid>
          ))}
        </Grid>
      </Box>
    </RenderController>
  );
};
