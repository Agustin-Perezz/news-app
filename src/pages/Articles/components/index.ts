export * from './Article';
export * from './ListArticles';
