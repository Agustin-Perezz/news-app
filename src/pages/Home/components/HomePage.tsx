import { Welcome } from '.';

const HomePage: React.FC = () => {
  return <Welcome />;
};

export default HomePage;
