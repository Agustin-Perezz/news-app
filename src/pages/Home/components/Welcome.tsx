import { Box, Typography } from '@mui/material';
import { useContext } from 'react';
import { ListCards } from '../../../components/ui/Cards';
import { NewsContext } from '../../../context/NewsContext';

export const Welcome: React.FC = () => {
  const { data, isLoading, isError } = useContext(NewsContext);
  return (
    <Box
      sx={{
        marginTop: '30px',
        display: 'flex',
        justifyContent: 'center',
        flexWrap: 'wrap',
      }}
    >
      <Typography
        sx={{ marginBottom: '20px', width: '100%', textAlign: 'center' }}
        variant="h4"
      >
        Welcome.
      </Typography>
      <ListCards
        dataArticles={data}
        isLoading={isLoading}
        isError={isError}
        titleText="More recent top news."
      />
    </Box>
  );
};
