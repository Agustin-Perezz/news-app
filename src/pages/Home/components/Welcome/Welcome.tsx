import { Box, Typography } from '@mui/material';
import { useContext } from 'react';
import { ListCards } from '../../../../components/ui/Cards';
import { CacheContext } from '../../../../context/CacheContext';
import { NewsContext } from '../../../../context/NewsContext';
import { ModalCountries } from '../Modal';

export const Welcome: React.FC = () => {
  const { data, isLoading, isError } = useContext(NewsContext);
  const { cache } = useContext(CacheContext);

  return (
    <Box
      sx={{
        marginTop: '30px',
        display: 'flex',
        justifyContent: 'center',
        flexWrap: 'wrap',
      }}
    >
      <ModalCountries />
      <Typography sx={{ marginBottom: '20px', width: '100%', textAlign: 'center' }} variant="h4">
        Welcome.
      </Typography>
      <ListCards
        dataArticles={data}
        isLoading={isLoading}
        isError={isError}
        headerTitle={cache.country ? `Top news from ${cache.country}.` : `Most recent top news.`}
      />
    </Box>
  );
};
