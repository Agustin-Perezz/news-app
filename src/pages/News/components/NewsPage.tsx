import { Box } from '@mui/material';
import { useContext } from 'react';
import { ListCards } from '../../../components/ui/Cards';
import { NewsContext } from '../../../context/NewsContext';
import { Header } from './Header';

const NewsPage: React.FC = () => {
  const { data, isLoading, isError } = useContext(NewsContext);
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: '30px',
      }}
    >
      <Header headerTitle="Results for search." />
      <ListCards dataArticles={data} isLoading={isLoading} isError={isError} />
    </Box>
  );
};

export default NewsPage;
