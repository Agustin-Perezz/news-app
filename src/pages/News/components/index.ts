export * from '../../../components/ui/Cards/GeneralCard';
export * from '../../../components/ui/Cards/ListCards';
