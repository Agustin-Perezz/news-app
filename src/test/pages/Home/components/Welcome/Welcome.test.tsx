import { screen } from '@testing-library/react';
import { Welcome } from '../../../../../pages/Home/components/Welcome';
import { renderWhitContext } from '../../../../utils';

type Props = { cache: Record<string, string> };

describe('Test in <Welcome />', () => {
  function setup({ cache }: Props) {
    return renderWhitContext({
      children: <Welcome />,
      pathname: '/home',
      cache,
    });
  }

  it('should match snapshot', () => {
    const { container } = setup({ cache: {} });
    expect(container).toMatchSnapshot();
  });
  it('should render main content', () => {
    setup({ cache: {} });
    screen.getByText('Welcome.');
    screen.getByText('Most recent top news.');
    screen.findByTestId('modal');
  });
  it('should render correctly information if the cache contain any country', () => {
    setup({ cache: { country: 'Argentina' } });
    screen.getByText('Top news from Argentina.');
    expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
  });
});
