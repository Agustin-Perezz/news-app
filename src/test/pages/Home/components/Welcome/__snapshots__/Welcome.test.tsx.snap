// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Welcome /> should match snapshot 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiBox-root css-14hrs0g"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-1r4f7b3-MuiTypography-root"
    >
      Welcome.
    </h4>
    <div>
      <div
        class="MuiGrid-root css-87ggms-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-1b9hrwo"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
          >
            Most recent top news.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-z7x17r-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-bzgqu1-MuiGrid-root"
            style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
